src/pages/campaigns/donation-v2/[slug]/index.tsx

```diff
@@ -0,0 +1,52 @@
+import { GetServerSideProps, GetServerSidePropsContext } from 'next'
+import { serverSideTranslations } from 'next-i18next/serverSideTranslations'
+import { QueryClient, dehydrate } from '@tanstack/react-query'
+import Stripe from 'stripe'
+import { AxiosResponse } from 'axios'
+
+import { apiClient } from 'service/apiClient'
+import { endpoints } from 'service/apiEndpoints'
+import { queryFnFactory } from 'service/restRequests'
+import { CampaignResponse } from 'gql/campaigns'
+
+import { Currencies } from 'components/withdrawals/WithdrawalTypes'
+import DonationFlowPage from 'components/donation-flow/DonationFlowPage'
+
+export const getServerSideProps: GetServerSideProps = async (ctx: GetServerSidePropsContext) => {
+  const { slug } = ctx.query
+  const client = new QueryClient()
+  await client.prefetchQuery<CampaignResponse>(
+    [endpoints.campaign.viewCampaign(String(slug)).url],
+    queryFnFactory<CampaignResponse>(),
+  )
+  await client.prefetchQuery(
+    [endpoints.donation.singlePrices.url],
+    queryFnFactory<Stripe.Price[]>(),
+  )
+
+  //create and prefetch the payment intent
+  const { data: paymentIntent } = await apiClient.post<
+    Stripe.PaymentIntentCreateParams,
+    AxiosResponse<Stripe.PaymentIntent>
+  >(endpoints.donation.createPaymentIntent.url, {
+    amount: 100,
+    currency: Currencies.BGN,
```

Comment:
Adding a note here to keep in mind we have hardcoded values for the amount.

Reply:
Yeah this value is hardcoded, because this is the initial creation of the intent. 
Stripe doesn't let us create a PaymentIntent without a value, because their assumption is that you would be creating the payment intent after the amount has been chosen. And in our case we need it beforehand.

Reply:
That's an unusual flow for the payment intents, however, we need to document it so the purpose is clear in our project.

Reply:
We might also pick a value that's closer to the initial values that we offer to the user. In this way updating the PI won't be necessary if they really pick the default value.

Reply:
> We might also pick a value that's closer to the initial values that we offer to the user. In this way updating the PI won't be necessary if they really pick the default value.

We can do that if we decide to have one of the values preselected.